parser/src/ast.rs

```diff
@@ -47,7 +47,7 @@ pub enum Statement {
     Break,
     Continue,
     Return {
-        value: Option<Vec<Expression>>,
+        value: Option<Box<Expression>>,
```

Comment:
couldn't this be simply an `Option<Expression>`?

Reply:
Maybe, it's mainly done due to memory size of the different enum variants that `Box` is present. Maybe `Option` also acts as a sort of Box like pointer to reduce the full size of `Return`.

Reply:
Well, other Statement variants already hold a non-boxed Expression, so that shouldn't matter here. IMO either all or none (unless later performance measurements prove otherwise).